src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -0,0 +1,236 @@
+                self.numa = ''.join(open(p, 'r').read()).rstrip()
+
+    def set_providers(self, dev_name, prefix):
+        """Get the provider information."""
+        # Setup and run command
+        fi_info = os.path.join(prefix, "bin", "fi_info -d {}".format(dev_name))
+        out = process.run(fi_info)
+
+        # Parse the list and divide the info
+        # pylint: disable=no-member
+        prov = re.findall(
+            r"(?:provider:|domain:)\s+([A-Za-z0-9;_+]+)", out.stdout, re.M)
+        # pylint: enable=no-member
+        info = [prov[i:(i + 2)] for i in range(0, len(prov), 2)]
+
+        # Get providers that are supported
+        supported = []
+        for i in info:
+            for sup in self.SUPPORTED_PROV:
+                if sup in i[0] and "rxd" not in i[0]:
+                    supported.append(i)
+
+        # Check that the domain name is in output found.
+        providers = []
+        for i in supported:
+            if i[1] == dev_name:
+                providers.append(i[0])
+
+        if self.providers:
+            self.providers.extend(providers)
+        else:
+            self.providers = providers
+
+
+class DmgNetworkScanTest(ControlTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+    Simple test to verify the network scan function of the dmg tool.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DmgNetworkScanTest object."""
+        super(DmgNetworkScanTest, self).__init__(*args, **kwargs)
+        self.setup_start_agents = False
+
+    @staticmethod
+    def get_devs():
+        """ Get list of devices."""
+        devs = {}
+        for dev in os.listdir("/sys/class/net/"):
+            devs[dev] = [dev]
+            # Check if we need to add infiniband dev to list
+            p = "/sys/class/net/{}/device/infiniband".format(dev)
+            if os.path.exists(p):
+                devs[dev].extend(os.listdir(p))
+        return devs
+
+    @staticmethod
+    def clean_info(net_devs):
+        """Clean up the devices found in the system.
+
+        Clean up devices that don't have a numa node and some providers that
+        might be duplicates, also need to attach 'ofi' to provider name.
+
+        Args:
+            net_devs (NetDev): list of NetDev objects.
+
+        Returns:
+            NetDev: list of NetDev objects.
+
+        """
+        idx_l = []
+        for idx, dev in enumerate(net_devs):
+            if dev.f_iface == "lo":
+                idx_l.append(idx)
+            elif not dev.numa:
+                idx_l.append(idx)
+            else:
+                if dev.providers:
+                    no_dups = list(dict.fromkeys(dev.providers))
+                    dev.providers = ["ofi+" + i for i in no_dups]
+        _ = [net_devs.pop(idx - i) for i, idx in enumerate(sorted(idx_l))]
+
+        return net_devs
+
+    def get_sys_info(self, host):
+        """Get the system device information."""
```

Comment:
Better to have parameter and return value description.

Reply:
Yes, will do.